ZenPacks/zenoss/Microsoft/Windows/modeler/plugins/zenoss/winrm/WinMSSQL.py

```diff
@@ -197,42 +197,76 @@ def __init__(self, conn_info, log):
         write-host $results_json;
     '''
 
-    ALL_SQL_INSTANCES_PS_SCRIPT = '''
-        <# Get registry key for instances (with 2003 or 32/64 Bit 2008 base key) #>
-        $reg = [Microsoft.Win32.RegistryKey]::OpenRemoteBaseKey('LocalMachine', $hostname);
-        $baseKeys = 'SOFTWARE\Microsoft\Microsoft SQL Server', 'SOFTWARE\Wow6432Node\Microsoft\Microsoft SQL Server';
-        foreach ($regPath in $baseKeys) {
-            $regKey= $reg.OpenSubKey($regPath);
-            If ($regKey -eq $null) {Continue};
-
-            <# Get installed instances' names (both cluster and local) #>
-            If ($regKey.GetSubKeyNames() -contains 'Instance Names') {
-                $regKey = $reg.OpenSubKey($regpath+'\Instance Names\SQL');
-                $instances = @($regkey.GetValueNames());
-            } ElseIf ($regKey.GetValueNames() -contains 'InstalledInstances') {
-                $instances = $regKey.GetValue('InstalledInstances');
-            } Else {Continue};
-
-            $instances_list = New-Object System.Collections.Arraylist;
-            $instances | % {
-                $instanceValue = $regKey.GetValue($_);
-                $instanceReg = $reg.OpenSubKey($regpath+'\\'+$instanceValue);
-                $version = $instanceReg.OpenSubKey('MSSQLServer\CurrentVersion').GetValue('CurrentVersion');
-                $instance_ver = $_;$instance_ver+=':';$instance_ver+=$version;
-                <#Insert info about each SQL Instance into dictionary#>
-                $instance_info = New-Object 'system.collections.generic.dictionary[string,string]';
-                $instance_info['name'] = $_;
-                $instance_info['version'] = $version;
-                $instances_list.Add($instance_info) > $null;
-            };
-            break;
-        };
-        $result = New-Object 'system.collections.generic.dictionary[string, object]';
-        $result['hostname'] = hostname;
-        $result['instances'] = $instances_list;
-        $result_in_json = ConvertTo-Json $result;
-        write-host $result_in_json
-    '''
+    ALWAYS_ON_RESOURCES_ON_NODE_PS_SCRIPT = (
```

Comment:
Add Replicas and SQL Instances info collection to have this info on node level (before accessing SQL Instance directly as the may be inaccessible).